tests/std/tests/Dev11_0235721_async_and_packaged_task/test.cpp

```diff
@@ -260,12 +273,42 @@ void test_shared_future_noexcept_copy_impl() {
     assert(!copyCtord.valid());
 }
 
-void test_shared_future_noexcept_copy() {
+void test_shared_future_noexcept() {
+    test_future_shared_future_noexcept_impl<int>();
+    test_future_shared_future_noexcept_impl<int&>();
+    test_future_shared_future_noexcept_impl<void>();
+
     test_shared_future_noexcept_copy_impl<int>();
     test_shared_future_noexcept_copy_impl<int&>();
     test_shared_future_noexcept_copy_impl<void>();
 }
 
+// Also test the non-constructibility of future from (future, {}) and (shared_future, {})
+template <typename Void, typename T, typename... Args>
+constexpr bool is_constructible_with_trailing_empty_brace_impl = false;
+
+template <typename T, typename... Args>
+constexpr bool
+    is_constructible_with_trailing_empty_brace_impl<void_t<decltype(T(declval<Args>()..., {}))>, T, Args...> = true;

```

Comment:
Test nitpick: When implementing custom type traits, it's a good idea for the test to have at least one positive and one negative case. This prevents the type trait itself from being damaged in a way that always returns `true` or `false`. Here, all of the usage is verifying `!is_constructible_with_trailing_empty_brace`, so we should add a positive case - `pair` would suffice (it actually has `<class U1 = T1, class U2 = T2>` specifically to handle empty braces).